src/Raven.Server/Documents/Indexes/IndexDefinitionBase.cs

```diff
@@ -229,21 +235,27 @@ private static unsafe void EncryptStream(StorageEnvironmentOptions options, Memo
         private static unsafe void DecryptStream(StorageEnvironmentOptions options, MemoryStream stream)
         {
             var buffer = stream.ToArray();
-            var nonce = new byte[sizeof(long)];
+            var nonce = new byte[Sodium.crypto_aead_xchacha20poly1305_ietf_npubbytes()];
             var data = new byte[buffer.Length - nonce.Length];
 
             Array.Copy(buffer, 0, data, 0, buffer.Length - nonce.Length);
             Array.Copy(buffer, buffer.Length - nonce.Length, nonce, 0, nonce.Length);
 
-            var decryptedData = new byte[data.Length - Sodium.crypto_aead_chacha20poly1305_ABYTES()];
+            var decryptedData = new byte[data.Length - Sodium.crypto_aead_xchacha20poly1305_ietf_abytes()];
 
+            fixed (byte* ctx = Sodium.Context)
             fixed (byte* pData = data)
             fixed (byte* pDecryptedData = decryptedData)
             fixed (byte* pNonce = nonce)
             fixed (byte* pKey = options.MasterKey)
             {
+                var subKey = stackalloc byte[32];
+                ulong subKeyId = 1;
```

Comment:
You are trying to decrypt with a _different_ subkeyId_, that isn't going to work.
Please also add a test for this.